core/unit_test/TestLocalDeepCopy.hpp

```diff
@@ -1042,4 +1042,73 @@ TEST(TEST_CATEGORY, local_deepcopy_rangepolicy_layoutright) {
   }
 }
 #endif
+
+namespace Impl {
+template <typename T, typename SHMEMTYPE>
+using ShMemView =
+    Kokkos::View<T, Kokkos::LayoutRight, SHMEMTYPE, Kokkos::MemoryUnmanaged>;
+
+struct DeepCopyScratchFunctor {
+  DeepCopyScratchFunctor(
+      Kokkos::View<double*, TEST_EXECSPACE::memory_space> check_view_1,
+      Kokkos::View<double*, TEST_EXECSPACE::memory_space> check_view_2)
+      : check_view_1_(check_view_1),
+        check_view_2_(check_view_2),
+        N_(check_view_1.extent(0)) {}
+
+  KOKKOS_INLINE_FUNCTION void operator()(
+      Kokkos::TeamPolicy<TEST_EXECSPACE,
+                         Kokkos::Schedule<Kokkos::Dynamic>>::member_type team)
+      const {
+    using ShmemType = TEST_EXECSPACE::scratch_memory_space;
+    auto shview =
+        Impl::ShMemView<double**, ShmemType>(team.team_scratch(1), N_, 1);
+
+    Kokkos::parallel_for(
+        Kokkos::TeamThreadRange(team, N_), KOKKOS_LAMBDA(const size_t& index) {
+          auto thread_shview = Kokkos::subview(shview, index, Kokkos::ALL());
+          Kokkos::Experimental::local_deep_copy(thread_shview, index);
+        });
+    Kokkos::Experimental::local_deep_copy(
+        check_view_1_, Kokkos::subview(shview, Kokkos::ALL(), 0));
+
+    Kokkos::Experimental::local_deep_copy(team, shview, 6.);
+    Kokkos::Experimental::local_deep_copy(
+        check_view_2_, Kokkos::subview(shview, Kokkos::ALL(), 0));
+  }
+
+  Kokkos::View<double*, TEST_EXECSPACE::memory_space> check_view_1_;
+  Kokkos::View<double*, TEST_EXECSPACE::memory_space> check_view_2_;
+  int const N_;
+};
+}  // namespace Impl
+
+TEST(TEST_CATEGORY, deep_copy_scratch) {
+  using TestDeviceTeamPolicy = Kokkos::TeamPolicy<TEST_EXECSPACE>;
+
+  const int N                = 8;
+  const int bytes_per_team   = N * sizeof(double),
+            bytes_per_thread = N * sizeof(double);
```

Comment:
This isn't correct. You should ask the `ShMemView<double**,ShmemType>` what its scratch requirement are.

Reply:
Also why do you need the per thread scratch?